lib/Extractor/KLEEBuilder.cpp

```diff
@@ -136,8 +140,25 @@ ref<Expr> ExprBuilder::makeSizedArrayRead(unsigned Width, StringRef Name,
       new Array(ArrayNames.makeName(NameStr), 1, 0, 0, Expr::Int32, Width));
   ArrayVars.push_back(Origin);
 
+  std::vector<unsigned> ZeroBits, OneBits;
   UpdateList UL(Arrays.back().get(), 0);
-  return ReadExpr::create(UL, klee::ConstantExpr::alloc(0, Expr::Int32));
+  ref<Expr> Var = ReadExpr::create(UL, klee::ConstantExpr::alloc(0, Expr::Int32));
+  if (Origin && Origin->K == Inst::Var && (Origin->KnownZeros.getBoolValue() ||
+                                           Origin->KnownOnes.getBoolValue())) {
+    ref<Expr> KnownZerosExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    ref<Expr> KnownOnesExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    // KnownZeros
```

Comment:
you can drop these comments, it's obvious that the following code is processing zeros and then ones